src/sys.rs

```diff
@@ -69,15 +69,18 @@ mod common {
     }
 }
 
-#[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
+#[cfg(target_os = "wasi")]
```

Comment:
Please pardon my lack of familiarity here, but...: The `wasm32` and `wasi` targets are not synonymous, correct?  If that's the case, why are you replacing the definition for `wasm32`, rather than adding an implementation for `wasi`?

Reply:
The previous `wasm32` without `emscripten` implementations were just stubs panic'ing with `unimplemented!()`.

That code was probably written before `wasi` existed. It is impossible to retrieve the current time in WebAssembly without WASI. If we try to compile for base-metal `wasm32`, compilation should fail rather than producing a package that will panic at runtime.

Reply:
I agree that a compile-time error is preferred to one at runtime.  That said, unless it's impossible that there are no `wasm32` consumers (perhaps using the crate for other utility purposes), I don't feel comfortable pulling the definition.

Reply:
Stubs restored.